jgiven-core/src/main/java/com/tngtech/jgiven/impl/inject/ValueInjector.java

```diff
@@ -90,6 +90,8 @@ public void updateValues( Object object ) {
                 if( value != null ) {
                     field.set( object, value );
                     log.debug( "Setting field {} to value {}", field, value );
+                } else if( isRequired( field ) ) {
```

Comment:
Well, in general I agree, however, JGiven is already doing a lot of reflection stuff, so I like to keep it at a minimum if possible. But I guess it will not be measurable.